arch/x86: Generalize cache manipulation exposure

tbursztyka opened the pull request:
Generalize cache exposure from what used to be in ia32 only before.

andyross commented on arch/x86/core/intel64/cache_s.S:
```diff
@@ -0,0 +1,74 @@
+	movq %rdx, %rax
+	andq $CPUID_CFLSH_BIT, %rax
+	popq %rbx
+	ret
```
Mild complaint: runtime feature detection isn't really a Zephyr thing.  Obviously x86 is right at the boundary where sometimes you have to, but in this case?  I dunno, cache line size is 512 bytes across the whole universe now and CLFLUSH works everywhere too.  I honestly think you could skip this, and if we really need it across architectures we have in-tree it belongs in devicetree instead (and I say that as someone who really hates devicetree and loves x86 assembly!)

tbursztyka replied:
Ok that means updating ia32 as well then (I just c/p cache_s.S from there with minor modifications)

andyross reviewed (changes requested):
Agree with the API motion, mildly opposed to detection code.  But I'm willing to be overruled.

Commit:
dts/x86: Add cache line size

There is a property for it so let's use it.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
arch/x86: Generalize cache manipulation functions

We assume that all x86 CPUs do have clflush instructions.
And the cache line size is now provided through DTS.

So detecting clflush instruction as well as the cache line size is no
longer required at runtime and thus removed.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
dts/bindings: Fixing x86 CPU compatibles by providing proper yaml files

Make compatible name following the rule as well.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

andyross reviewed (approved):
Sorry, I didn't see your reply until you had already made the change.  I didn't realize this was a pre-existing API.   Still, I like this version better anyway (but would be fine with original too)

Commit:
boards/x86: Give proper board compatible names

These compatible describe the vendor/board, not the cpu/model.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
